### PR TITLE
feat(cart): add an action that payment modules can use to add payment info

### PR DESCRIPTION
**Pull request by lderrickable**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information
Comments were added to explain why some of these changes are necessary. Should be temporary.

### CODE REVIEW

File: libs/cart/src/models/cart-payment.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
 export interface DaffCartPaymentMethod {
-  method: string;
+	method: string;
+	paymentToken?: string;
```
Comment (griest024) [inline libs/cart/src/models/cart-payment.ts:3]: Doesn't this need to be an object to accomodate the [extra payment data](https://devdocs.magento.com/guides/v2.3/graphql/payment-methods/authorize-net.html#additional-payment-information)?

Comment (lderrickable) [reply libs/cart/src/models/cart-payment.ts:3]: That's deprecated. We're going to use an authorizenet extension or something

Comment (lderrickable) [reply libs/cart/src/models/cart-payment.ts:3]: Honest answer is though, I don't know

Comment (griest024) [reply libs/cart/src/models/cart-payment.ts:3]: ah well I did see the deprecation notice....okay I guess for now we'll just assume that a payment token is sufficient

Comment (lderrickable) [reply libs/cart/src/models/cart-payment.ts:3]: I'll try to figure out more

File: libs/cart/src/reducers/cart-payment-methods/cart-payment-methods.reducer.ts [hunk 2]
```diff
@@ -35,7 +35,24 @@ export function cartPaymentMethodsReducer<T extends DaffCart>(
...
+		case DaffCartPaymentMethodsActionTypes.CartPaymentMethodAddAction:
+			return {
+				...state,
+				cart: {
+					...state.cart,
+					payment: {
+						...action.payload
+					}
+				}
+			};
```
[the remark this answers is not in the document]
Comment (lderrickable) [reply libs/cart/src/reducers/cart-payment-methods/cart-payment-methods.reducer.ts]: done

File: libs/cart/src/models/cart-payment.ts (hunk 1 above)
Comment (lderrickable) [reply libs/cart/src/models/cart-payment.ts:3]: I found (I think) the extension, but it looks like it only supports up to magento 2.3: https://marketplace.magento.com/authorizenet-magento-module-authorizenet.html

I'm talking to Damien about it. Maybe I'll switch to paypal

Comment (lderrickable) [reply libs/cart/src/models/cart-payment.ts]: ok ready for review again

File: libs/cart/src/reducers/cart-payment/cart-payment.reducer.spec.ts [hunk 3]
```diff
@@ -238,4 +239,27 @@ describe('Cart | Reducer | Cart Payment', () => {
       expect(result.errors[DaffCartErrorType.Payment].length).toEqual(2);
     });
   });
+
+  describe('when CartPaymentMethodAddAction is triggered', () => {
+    let result;
+		let state: DaffCartReducerState<DaffCart>;
+		const stubPayment = {
+			method: 'method',
+			paymentToken: 'paymentToken'
```
Comment (griest024) [inline libs/cart/src/reducers/cart-payment/cart-payment.reducer.spec.ts:248]: Should this be `payment_info`?

Comment (lderrickable) [reply libs/cart/src/reducers/cart-payment/cart-payment.reducer.spec.ts]: yea, fixed

**Pull request merged**